File: an/Mingrisoft/08/03/Drawable Resource/src/main/java/com/mingrisoft/MainActivity.java
package com.mingrisoft;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //获取密码编辑框
        final EditText editText = (EditText) findViewById(R.id.editText);
        //获取登录按钮
        final Button button = (Button) findViewById(R.id.btn_login);
        //为编辑框设置监听事件
        editText.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
            }
            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                if (editText.length() > 0) {            // 判断编辑框内输入文字时
                    button.setEnabled(true);            //登录按钮为可用状态
                } else {                                //编辑框内没有文字时
                    button.setEnabled(false);           //登录按钮为不可用状态
                }
            }
            @Override
            public void afterTextChanged(Editable s) {
            }
        });


    }
}

File: an/MusicPlayer/app/src/main/java/com/example/musicplayer/MainActivity.kt
package com.example.musicplayer

import android.app.Activity
import android.media.MediaPlayer
import android.os.Bundle
import android.util.Log
import android.view.View
import android.view.WindowManager
import android.widget.ImageButton
import android.widget.Toast
import java.io.File

class MainActivity : Activity() {

    private lateinit var file : File
    private var player : MediaPlayer = MediaPlayer()
    private var isPause = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN)
        val btn_play = findViewById<ImageButton>(R.id.btn_play)
        val btn_stop = findViewById<ImageButton>(R.id.btn_stop)

        file = File("/sdcard/Music/Red Highheel.mp3")
        if (!file.exists()) {
            Toast.makeText(this, "Music file does not exist!", Toast.LENGTH_SHORT).show()
            return
        }
        player.setDataSource(file.absolutePath)
        player.prepare()

        player.setOnCompletionListener { MediaPlayer.OnCompletionListener { mediaPlayer ->
            play()
        } }

        btn_play.setOnClickListener(View.OnClickListener {  view ->
            if (player.isPlaying && !isPause) {
                player.pause()
                isPause = true
                btn_play.setImageResource(R.drawable.play)
            } else {
                player.start()
                isPause = false
                btn_play.setImageResource(R.drawable.pause)
            }
        })

        btn_stop.setOnClickListener(View.OnClickListener { view ->
            player.stop()
            btn_play.setImageResource(R.drawable.play)
            player.prepare()
        })
    }

    private fun play() {
        try {
            player.reset()
            player.setDataSource(file.absolutePath)
            player.prepare()
            player.start()
        } catch (e : Exception) {
            e.printStackTrace()
        }
    }

    override fun onDestroy() {
        if (player.isPlaying) {
            player.stop()
        }
        player.release()
        super.onDestroy()
        Log.i("Danny", "On Destory called")
    }

    override fun onStop() {
        super.onStop()
        Log.i("Danny", "On stop called")
    }
}
